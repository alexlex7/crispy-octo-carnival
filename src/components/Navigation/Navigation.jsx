import React, { useState, useEffect, useContext } from 'react';
import Container from '../Container/Container';
import NavList from '../NavList/NavList';
import Logo from '../Logo/Logo';
import Button from '../Button/Button';
import { useMediaQuery } from 'react-responsive';
import styles from './Navigation.module.css';
import PropTypes from 'prop-types';
import { ReactComponent as FavoriteIcon } from '../../assets/images/heart.svg';

export default function Navigation({ items, products = [] }) {
  const [isOpenMenu, setIsOpenMenu] = useState(false);
  const isMobile = useMediaQuery({ maxWidth: 1204 });

  useEffect(() => {
    setIsOpenMenu(!isMobile);
  }, [isMobile]);

  const handleClickMenu = () => {
    setIsOpenMenu(!isOpenMenu);
  };

  const totalFavorite = products.filter(
    (item) => item.isFavorite === true
  ).length;

  return (
    <Container>
      <div className={styles.wrapper}>
        <div className={styles.logoWrapper}>
          <Logo />
        </div>
        {isOpenMenu && (
          <div className={styles.navWrapper}>
            <NavList pages={items} />

            <div className={styles.container}>
              <div className={styles.favoriteWrapper}>
                <FavoriteIcon className={styles.favoriteIcon} />
                <p className={styles.favoriteText}>
                  {totalFavorite === 0 ? '' : totalFavorite}
                </p>
              </div>
              <Button type="search" />
              <Button type="menu" />
            </div>
          </div>
        )}
        {isMobile && <Button type="mobile menu" onClick={handleClickMenu} />}
      </div>
    </Container>
  );
}

Navigation.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      to: PropTypes.string.isRequired,
    })
  ),
};
