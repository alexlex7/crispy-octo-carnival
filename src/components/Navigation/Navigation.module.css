.favoriteWrapper {
  position: relative;
  width: 30px;
  height: 30px;

  display: flex;
  align-items: center;
  justify-content: center;
}

.favoriteIcon {
  width: 24px;
  height: 24px;
  fill: #244d4d;
}
.favoriteText {
  position: absolute;
  right: -5px;
  bottom: -5px;
}

@media screen and (max-width: 1204px) {
  .wrapper {
    padding: 40px 0px 40px 0px;
    position: relative;
    display: flex;
    justify-content: space-between;
  }
  .navWrapper {
    padding: 15px;

    position: absolute;
    top: 116px;
    right: 0px;
    display: flex;
    flex-direction: column;
    align-items: center;
    align-content: center;

    background-color: #fefafa;
    box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
  }

  .container {
    padding-top: 15px;
    display: flex;
  }
  .favoriteWrapper {
    margin-right: 8px;
  }
}

@media screen and (min-width: 1205px) {
  .wrapper {
    padding: 44px 0px 27px 0px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    align-content: center;
  }
  .logoWrapper {
    flex-grow: 3;
  }

  .navWrapper {
    display: flex;
    justify-content: space-between;
    align-items: center;
    align-content: center;
    flex-grow: 5;
  }
  .container {
    display: flex;
    align-items: center;
    column-gap: 18px;
  }
}
