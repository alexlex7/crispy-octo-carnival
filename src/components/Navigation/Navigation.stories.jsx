import React from 'react';

import Navigation from './Navigation';
import { pages } from '../../App';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Navigation/Navigation',
  component: Navigation,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <Navigation {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  items: pages,
};
