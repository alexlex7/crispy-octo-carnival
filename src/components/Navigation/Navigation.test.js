import { render } from '@testing-library/react';
import Navigation from './Navigation';
import pages from '../../mock/pages.json';
import { BrowserRouter } from 'react-router-dom';
test('renders navigation component', () => {
  console.log(pages);
  const { asFragment } = render(<Navigation items={pages} />, {
    wrapper: BrowserRouter,
  });

  expect(asFragment()).toMatchSnapshot();
});
