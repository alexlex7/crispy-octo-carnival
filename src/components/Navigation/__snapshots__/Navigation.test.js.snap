// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navigation component 1`] = `
<DocumentFragment>
  <div
    class="container "
  >
    <div
      class="wrapper"
    >
      <div
        class="logoWrapper"
      >
        <p
          class="logo"
        >
          DudeShape
        </p>
      </div>
      <div
        class="navWrapper"
      >
        <nav>
          <ul
            class="list"
          >
            <li
              class="listItem"
            >
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="listItem"
            >
              <a
                class="item"
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="listItem"
            >
              <a
                class="item"
                href="/features"
              >
                Features
              </a>
            </li>
            <li
              class="listItem"
            >
              <a
                class="item"
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="container"
        >
          <div
            class="favoriteWrapper"
          >
            <svg
              class="favoriteIcon"
            >
              heart.svg
            </svg>
            <p
              class="favoriteText"
            />
          </div>
          <button
            class="btn"
            type="button"
          >
            <svg
              class="icon"
            >
              searchIcon.svg
            </svg>
          </button>
          <button
            class="btn"
            type="button"
          >
            <svg
              class="icon"
            >
              menu.svg
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
